v1/src/test/java/com/google/cloud/teleport/templates/BigtableToParquetIT.java

```diff
@@ -0,0 +1,114 @@
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.avro.generic.GenericRecord;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Integration test for {@link BigtableToParquet} classic template. */
+@Category(TemplateIntegrationTest.class)
+@TemplateIntegrationTest(BigtableToParquet.class)
+@RunWith(JUnit4.class)
+public class BigtableToParquetIT extends TemplateTestBase {
+
+  private BigtableResourceManager bigtableResourceManager;
+
+  @Before
+  public void setUp() throws IOException {
+    bigtableResourceManager =
+        BigtableResourceManager.builder(testName, PROJECT)
+            .setCredentialsProvider(credentialsProvider)
+            .build();
+  }
+
+  @After
+  public void tearDown() {
+    ResourceManagerUtils.cleanResources(bigtableResourceManager);
+  }
+
+  @Test
+  public void testBigtableToParquet() throws IOException {
+    // Arrange
+    String tableId = generateTableId(testName);
+    bigtableResourceManager.createTable(tableId, ImmutableList.of("col1"));
+
+    long timestamp = System.currentTimeMillis() * 1000;
+    bigtableResourceManager.write(
+        ImmutableList.of(
+            RowMutation.create(tableId, "r1").setCell("col1", "c1", timestamp, "new-value1"),
+            RowMutation.create(tableId, "r2").setCell("col1", "c1", timestamp, "new-value2"),
+            RowMutation.create(tableId, "r3").setCell("col1", "c1", timestamp, "new-value3")));
+
+    PipelineLauncher.LaunchConfig.Builder options =
+        PipelineLauncher.LaunchConfig.builder(testName, specPath)
+            .addParameter("bigtableProjectId", PROJECT)
+            .addParameter("bigtableInstanceId", bigtableResourceManager.getInstanceId())
+            .addParameter("bigtableTableId", tableId)
+            .addParameter("outputDirectory", getGcsPath("output/"))
+            .addParameter("filenamePrefix", "bigtable-to-parquet-output-")
+            .addParameter("numShards", "1");
+
+    // Act
+    PipelineLauncher.LaunchInfo info = launchTemplate(options);
+    assertThatPipeline(info).isRunning();
+
+    PipelineOperator.Result result = pipelineOperator().waitUntilDone(createConfig(info));
+
+    // Assert
+    assertThatResult(result).isLaunchFinished();
+
+    List<Artifact> artifacts = gcsClient.listArtifacts("output/", Pattern.compile(".*"));
+    assertThat(artifacts).isNotEmpty();
+
+    List<GenericRecord> allRecords = new ArrayList<>();
+    for (Artifact artifact : artifacts) {
+      try {
+        allRecords.addAll(ParquetTestUtil.readRecords(artifact.contents()));
+      } catch (Exception e) {
+        throw new RuntimeException("Error reading " + artifact.name() + " as Parquet.", e);
+      }
+    }
+
+    assertThatGenericRecords(allRecords)
+        .hasRecordsWithStrings(ImmutableList.of("new-value1", "new-value2", "new-value3"));
```

Comment:
Nitpick: probably 

```
assertThatArtifacts(artifacts).asAvroRecords(SCHEMA)
     .hasRecordsWithStrings(ImmutableList.of("new-value1", "new-value2", "new-value3"));
```

Is a simpler construct here

Reply:
Personally, I would prefer that the test can make sure that the records can actually be converted to Parquet, instead of converting them to Avro, so I'm leaving this as is